File: ios/UniPark/UniPark/domain/entities/Scan.swift
import Foundation

public struct Scan: Identifiable, Codable, Sendable {
	public let id: UUID
	public let passId: UUID
	public let guardId: UUID
	public let lotId: UUID
	public let direction: ScanDirection
	public let scannedAt: Date
	public let idempotencyKey: String

	public init(
		id: UUID = UUID(),
		passId: UUID,
		guardId: UUID,
		lotId: UUID,
		direction: ScanDirection,
		scannedAt: Date,
		idempotencyKey: String
	) {
		self.id = id
		self.passId = passId
		self.guardId = guardId
		self.lotId = lotId
		self.direction = direction
		self.scannedAt = scannedAt
		self.idempotencyKey = idempotencyKey
	}
}

File: android/app/src/main/java/com/unipark/android/domain/entities/Scan.kt
package com.unipark.android.domain.entities

import android.os.Parcelable
import kotlinx.parcelize.Parcelize
import java.time.Instant
import java.util.UUID

@Parcelize
data class Scan(
    val id: UUID,
    val passId: UUID,
    val lotId: UUID,
    val guardId: UUID,
    val direction: ScanDirection,
    val status: ScanStatus,
    val scannedAt: Instant,
) : Parcelable

enum class ScanDirection { ENTRY, EXIT }

enum class ScanStatus { ACCEPTED, REJECTED, OFFLINE_PENDING }
